pytorch_lightning/trainer/trainer.py

```diff
@@ -1103,14 +1103,16 @@ def _run(
 
         verify_loop_configurations(self)
 
-        # hook
-        self._data_connector.prepare_data()
-
         # ----------------------------
         # SET UP TRAINING
         # ----------------------------
         self._call_callback_hooks("on_before_accelerator_backend_setup")
         self.strategy.setup_environment()
+
+        self.__setup_profiler()
+        # hook
+        self._data_connector.prepare_data()
```

Comment:
Is this change strictly necessary for the other changes in this PR to work? If not, then let's leave this for a separate PR. Hook order changes are important changes as they silently change the behaviour.

Reply:
Thanks for your suggestion. I remove this change and will submit it in another PR.